### PR TITLE
Ruby on Rails Group Project.

### PR DESCRIPTION
**Pull request by Favourezeugwa**
## In this project, I , @assadounto  and @RitobrotoMukherjee  have implemented the following;
- Followed the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
built with Devise gem.
- Food list:
displays a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
leads to a form that allows users to add new food.
- Recipes list:
displays a list of recipes created by the logged-in user as in the wireframe.
leads to recipe details.
If the user is the owner of the recipe, it allows the user to delete it.
- Public recipe list:
displays a list of all public recipes ordered by newest as in the wireframe.
leads to recipe details.
- Recipe details:
If the recipe is public or the user is the owner of the recipe, displays the recipe details as in the wireframe.
If the user is the owner of the recipe, should lead to the form that allows the user to add new food.

- Inventories list:
displays a list of inventories created by the logged-in user as in the wireframe.
 leads to inventory details.
If the user is the owner of the inventory, should allow to delete it.
- Inventory details:
 displays the inventory details as in the wireframe.
 leads to a form that allows users to add new food.
- Recipe details:
has a modal to choose an inventory to create a shopping list with, as in the wireframe.
- Inventory shopping list:
 shows the list of food that is missing by comparing the food in the recipe with the food in the inventory.
counts the total food items and total price of the missing food.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,105 @@
...
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Hat tip to [Microverse](https://bit.ly/MicroverseTN) for giving me this project to work on
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (MahmoudBakr23) [inline README.md:105]: ### Optional
- Please fix this link for a professional readme file

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    before { subject.save }
+
+    it 'name should be present' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'name should not be blank' do
+      subject.name = ''
+      expect(subject).to_not be_valid
```
Comment (MahmoudBakr23) [inline spec/models/user_spec.rb:16]: - [ ] Well done writing your Rspec tests so far. However, there are a few tests that haven't passed please fix them.
Git-Bksh__c_Users_mbakr_OneDrive_Desktop_Junk_Receipe-app-rails 10_15_2022 1_20_52 AM

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello team😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Great job working on the models and setting up the database according to the ERD
- Nice work with the validations of your models
- Well done on having the controllers maintained, optimized, detailed, and very organized
- Great work on the custom controller methods
- Nicely done setting up the routes
- Well done setting up the devise gem
- Nice and simple design
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_See the comments under the review._

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (RitobrotoMukherjee) [reply README.md]: Okay will update it

File: spec/models/user_spec.rb (hunk 2 above)
Comment (RitobrotoMukherjee) [reply spec/models/user_spec.rb]: Ah!! Sure, will fix it

Comment (RitobrotoMukherjee) [reply spec/models/user_spec.rb]: Ah!! Sure, will fix it

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED ✔️ 🟢 

## Hi team 👋, 

You have made a great effort by implementing the required changes 👏 
Your project is now complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by RitobrotoMukherjee** [comment]
@codepantha Thank you so much for approving our project

**Pull request merged**